bytecomp/translmod.ml

```diff
@@ -91,7 +91,10 @@ let rec apply_coercion loc strict restr arg =
         (fun _ -> apply_coercion loc Alias cc lam)
 
 and apply_coercion_field loc get_field (pos, cc) =
-  apply_coercion loc Alias cc (get_field pos)
+  if pos >= 0 then
+    apply_coercion loc Alias cc (get_field pos)
```

Comment:
Would it make sense to add `assert (pos >= 0)` in the definition of `get_field`?

Reply:
I've now moved the check itself into `get_field` which is slightly cleaner and more consistent with the other `get_field` function that appears in this file.